src/main/java/org/apache/mesos/hdfs/state/LiveState.java

```diff
@@ -46,19 +58,21 @@ public void removeStagingTask(final Protos.TaskID taskID) {
     stagingTasks.remove(taskID);
   }
 
-  public LinkedHashMap<Protos.TaskID, Protos.TaskStatus> getRunningTasks() {
+  public HashMap<String, Protos.TaskStatus> getRunningTasks() {
     return runningTasks;
   }
 
   public void removeRunningTask(Protos.TaskID taskId) {
-    if (isNameNode1Initialized()
+    if (!nameNode1TaskMap.isEmpty()
         && nameNode1TaskMap.keySet().iterator().next().getTaskId().equals(taskId)) {
       nameNode1TaskMap.clear();
-    } else if (isNameNode2Initialized()
+    } else if (!nameNode2TaskMap.isEmpty()
```

Comment:
Depending on the error state, these should be cleared even if they aren't initialized.